modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj

```diff
@@ -395,20 +393,53 @@
   [_]
   #{"sys" "INFORMATION_SCHEMA"})
 
-;; SQL Server doesn't support setting the holdability of an individual result set, otherwise this impl is basically
-;; the same as the default
-(defmethod sql-jdbc.execute/prepared-statement :sqlserver
-  [driver ^Connection conn ^String sql params]
-  (let [stmt (.prepareStatement conn sql
-                                ResultSet/TYPE_FORWARD_ONLY
-                                ResultSet/CONCUR_READ_ONLY)]
-    (try
-      (.setFetchDirection stmt ResultSet/FETCH_FORWARD)
-      (sql-jdbc.execute/set-parameters! driver stmt params)
-      stmt
-      (catch Throwable e
-        (.close stmt)
-        (throw e)))))
+;; From the dox:
+;;
+;; The ORDER BY clause is invalid in views, inline functions, derived tables, subqueries, and common table
+;; expressions, unless TOP, OFFSET or FOR XML is also specified.
+;;
+;; To fix this :
+;;
+;; - Remove `:order-by` without a corresponding `:limit` inside a `:join` (since it usually doesn't really accomplish
+;;   anything anyway; if you really need it you can always specify a limit yourself)
+;;
+;; - Add a max-results `:limit` to source queries if there's not already one
+
+(defn- in-source-query? [path]
+  (= (last path) :source-query))
```

Comment:
What happens if you pass in a path and it's `:source-query` in the middle in there somewhere, like you're getting a path deep inside some recondite source query?

Reply:
Same thought applies to `in-join-source-query?`

Reply:
I think I need to comment this a little better to clarify, but `in-source-query?` is only called at the same level `:limit` appears, so it's parent either *has* to be `:source-query` or just `:query` (if it's at the top level). 

For `in-join-source-query?` the additional constraint is that the parent source query is what we're joining against... you're right that we could be in a source query in a source query inside a join. In that case it would still be `in-source-query?` but *not* `in-join-source-query?`. 

Honestly I'm not 100% sure whether we would want to aggressively remove `:order-by` in that case as well. I'm inclined to leave it since it's working and it's possibly a dangerous assumption to make (if that third-level source query is itself joining some other tables or something there's a chance we might be changing the meaning of the query)

Reply:
I'll add some more comments detailing this